sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -93,7 +94,7 @@ private[spark] object SQLConf {
           isPublic: Boolean = true): SQLConfEntry[Int] =
       SQLConfEntry(key, defaultValue, { v =>
         try {
-          v.toInt
+          Utils.byteStringAsBytes(v).toInt
```

Comment:
Hello Sean: Thanks for your comment. Yes, you are right. There are other methods are not meaning to use memory sizes. (like COLUMN_BATCH_SIZE, etc).  There are couple approaches, can  you suggest which way is preferable way for this problem or suggest a new way to fix this ?
1. we will document that [g|G|m|M|k|K] means memory size.

2. create a new method of intConf for AUTO_BROADCASTJOIN_THRESHOLD.

3.  create a rule to the parseByteString. like K/KB/M/MB means 1024, k/kb/m/mb means 1000.

Thanks.